app/services/search_service/fuzzy_search/goods_nomenclature_query.rb

```diff
@@ -6,70 +6,82 @@ def query(query_opts = {})
-                            {
-                              bool: {
-                                must: [
-                                  { range: { validity_start_date: { lte: date } } },
-                                  { bool: { must_not: { exists: { field: 'validity_end_date' } } } },
-                                ],
-                              },
-                            },
-                            # or item has blank validity_start_date and validity_end_date
-                            {
-                              bool: {
-                                must: [
-                                  { bool: { must_not: { exists: { field: 'validity_start_date' } } } },
-                                  { bool: { must_not: { exists: { field: 'validity_end_date' } } } },
-                                ],
-                              },
-                            },
-                          ],
+                    },
+                  },
+                  multi_match_part(query_opts),
+                  {
+                    bool: {
+                      should: [
+                        # actual date is either between item's (validity_start_date..validity_end_date)
+                        {
+                          bool: {
+                            must: [
+                              { range: { validity_start_date: { lte: date } } },
+                              { range: { validity_end_date: { gte: date } } },
+                            ],
+                          },
                         },
-                      },
-                    ],
+                        # or is greater than item's validity_start_date
+                        # and item has blank validity_end_date (is unbounded)
+                        {
+                          bool: {
+                            must: [
+                              { range: { validity_start_date: { lte: date } } },
+                              { bool: { must_not: { exists: { field: 'validity_end_date' } } } },
+                            ],
+                          },
+                        },
+                        # or item has blank validity_start_date and validity_end_date
+                        {
+                          bool: {
+                            must: [
+                              { bool: { must_not: { exists: { field: 'validity_start_date' } } } },
+                              { bool: { must_not: { exists: { field: 'validity_end_date' } } } },
+                            ],
+                          },
+                        },
+                      ],
+                    },
                   },
-                },
+                ],
               },
             },
             size: INDEX_SIZE_MAX,
           },
         }
       end
+
+      def multi_match_part(query_opts = {})
+        query = {
+          # match the search phrase
```

Comment:
This comment is a lie - we'd want shingles if we wanted to match a phrase. Remove me